src/Forms/HTMLEditor/HTMLEditorField.php

```diff
@@ -123,13 +125,9 @@ public function getAttributes()
         );
     }
 
-    /**
-     * @param DataObject|DataObjectInterface $record
-     * @throws Exception
-     */
     public function saveInto(DataObjectInterface $record)
     {
-        if ($record->hasField($this->name) && $record->escapeTypeForField($this->name) != 'xml') {
+        if (!$this->usesXmlFriendlyField($record)) {
```

Comment:
This was the only usage of `escapeTypeForField` anywhere, so I've just pulled the necessary functionality into a private method instead.